cmake/platform.cmake

```diff
@@ -137,7 +137,13 @@ elseif(UNIX OR MINGW)
             append(CMAKE_CCXX_SANITIZER_FLAGS "-g -fno-omit-frame-pointer")
         endif()
     elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        set(DEF_ARCH_OPT_FLAGS "-msse4.1")
+        if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
+             set(DEF_ARCH_OPT_FLAGS "-O3 -mcpu=native")
+             set(DNNL_ENABLE_JIT_PROFILING "OFF")
```

Comment:
qq: since `DNNL_ENABLE_JIT_PROFILING` is option, shouldn't you use `CACHE BOOL "OFF" FORCE` in `set` to reset it?

Reply:
Fair point, done.